lib/wda-mac.js

```diff
@@ -282,21 +295,74 @@ class WDAMacServer {
     }
   }
 
+
+  /**
+   * @typedef {Object} ProxyProperties
+   *
+   * @property {string} scheme - The scheme proxy to.
+   * @property {string} host - The host name proxy to.
+   * @property {number} port - The port number proxy to.
+   * @property {string} path - The path proxy to.
+   */
+
+  /**
+   * Returns proxy information where WDAMacServer proxy to.
+   *
+   * @param {Object} caps - The capabilities in the session.
+   * @return {ProxyProperties}
+   */
+  parseProxyProperties (caps) {
+    let scheme = 'http';
+    if (!caps.webDriverAgentMacUrl) {
+      return {
+        scheme,
+        server: (this.process?.host ?? caps.systemHost) ?? DEFAULT_SYSTEM_HOST,
+        port: (this.process?.port ?? caps.systemPort) ?? DEFAULT_SYSTEM_PORT,
+        path: ''
+      };
+    }
+
+    const {protocol, hostname, port, pathname} = url.parse(caps.webDriverAgentMacUrl);
+    if (_.isString(protocol)) {
+      scheme = protocol.split(':')[0];
+    }
+    return {
+      scheme,
+      server: hostname ?? DEFAULT_SYSTEM_HOST,
+      port: _.isNil(port) ? DEFAULT_SYSTEM_PORT : _.parseInt(port),
+      path: pathname === '/' ? '' : pathname
+    };
+  }
+
   async startSession (caps) {
     this.serverStartupTimeoutMs = caps.serverStartupTimeout ?? this.serverStartupTimeoutMs;
-    const wasProcessInitNecessary = await this.process.init(caps);
 
-    if (wasProcessInitNecessary || !(await this.isProxyReady(false))) {
+    this.isProxyingToRemoteServer = !!caps.webDriverAgentMacUrl;
+
+    let wasProcessInitNecessary;
+    if (this.isProxyingToRemoteServer) {
+      wasProcessInitNecessary = false;
+    } else {
+      if (!this.process?.isRunning) { this.process = new WDAMacProcess(); }
```

Comment:
Please format the block to to multiple lines

Reply:
we should not create a new instance if the process is not running. consider changing the condition to `if(!this.process)...`